### PR TITLE
add rank warning

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?
Fixes # (issue).

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
now it is implemented only for model, but shall we doe rank warning also for callbacks?

**Comment by williamFalcon** [comment]
i   think it  needs to  be a  standalone  package.

```
from pl import warnings

warnings.rank_zero_warn()
```

don't make it so we have to instantiate  an instance... also instance doesn't make sense since we just need to call it  from wherever.

**Comment by williamFalcon** [comment]
Also, replace  all the  warnings across the  project with this.

**Comment by Borda** [comment]
but the rank is depending on the backend so you cannot use a global rank value... can you?

**Comment by Borda** [comment]
> warnings.rank_zero_warn()

but this has to have an extra parameter of the actual rank... 
so we can have warning called from where ever but not with rank specific

**Comment by williamFalcon** [comment]
warnings inside lightning are ALL rank specific.

File: pytorch_lightning/utilities/warnings.py [hunk 1]
```diff
@@ -2,25 +2,8 @@
...
-
-    @rank.setter
-    def rank(self, value: int) -> None:
-        """Set the process rank."""
-        self._rank = value
-
-    @rank_zero_only
-    def __call__(self, *args, **kwargs):
-        warnings.warn(*args, **kwargs)
+def rank_zero_warn(*args, trainer=None, **kwargs):
```
Comment (williamFalcon) [inline pytorch_lightning/utilities/warnings.py:6]: don't pass in trainer.

create another method to set the global rank in warnings

```python
rank_zero_warn.set_rank(k)
```

**Pull request merged**